pkg/workloads/helpers.go

```diff
@@ -107,8 +107,13 @@ func (wh *WorkloadHelper) Run(workload string, metricsProfiles []string, alertsP
 		log.Error(err)
 	}
 	wh.Metadata.Passed = rc == 0
-	if indexer != nil {
-		IndexMetadata(indexer, wh.Metadata)
+	for pos, indexer := range ConfigSpec.Indexers {
+		log.Infof("📁 Creating indexer: %s", indexer.Type)
+		idx, err := indexers.NewIndexer(indexer.IndexerConfig)
+		if err != nil {
+			log.Fatalf("Error creating indexer %d: %v", pos, err.Error())
+		}
+		IndexMetadata(idx, wh.Metadata)
```

Comment:
You don't need to create new indexers, they are not part of metricsScraper:

```suggestion
  for _, indexer := range metricsScraper.IndexerLis
		IndexMetadata(indexer, wh.Metadata)
```

Reply:
I think we can also remove the indexer variable

Reply:
Finished all my experiments and incorporated all the suggestions.